### PR TITLE
#85 add is_active into menu data

### PR DESCRIPTION
**Pull request by kiber**
DEV-65561

### CODE REVIEW

File: view/adminhtml/web/vue/field-type/checkbox.vue [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+                fieldId: '',
+                checkboxValue: this.value
+            }
+        },
+        mounted: function() {
+            this.fieldId = 'snowmenu_' + this.id + '_' + this._uid;
+        },
+        watch: {
+          checkboxValue: function(newValue, oldValue) {
+            this.item.is_active = newValue ? '1' : '0'
```
Comment (Igloczek) [inline view/adminhtml/web/vue/field-type/checkbox.vue:35]: We shouldn't mutate the prop directly. If it's just an UI thing move it to computed.

Comment (Januszpl) [reply view/adminhtml/web/vue/field-type/checkbox.vue:35]: Yes. I agree. But we would first to refactor few things related to saving whole menu data. and I think it should be done in a new ticket

File: Model/Menu/NodeRepository.php [hunk 2]
```diff
@@ -29,6 +38,11 @@ public function __construct(
         $this->searchResultsFactory = $searchResultsFactory;
     }
 
+    /**
+     * @param NodeInterface $object
+     * @return NodeInterface
+     * @throws CouldNotSaveException
+     */
```
Comment (jochhop) [inline Model/Menu/NodeRepository.php:45]: I would use `@inheritdoc` instead (I see it's used already in Menu.php model class, so let's keep the same format)

Comment (kiber) [reply Model/Menu/NodeRepository.php:43]: Added changes for all methods

**Review by mcjwsk — APPROVED** [review]

**Review by mcjwsk — APPROVED** [review]

**Pull request merged**